File: project/jaehyeon-onboarding/src/main/java/fc/icd/jaehyeononboarding/survey/controller/SurveyRestController.java
package fc.icd.jaehyeononboarding.survey.controller;

import fc.icd.jaehyeononboarding.common.model.NoDataResponse;
import fc.icd.jaehyeononboarding.survey.model.dto.SurveyCreateDTO;
import fc.icd.jaehyeononboarding.survey.model.dto.SurveyUpdateDTO;
import fc.icd.jaehyeononboarding.survey.service.SurveyService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Slf4j
@RequiredArgsConstructor
@RestController
public class SurveyRestController {

    private final SurveyService surveyService;

    @PostMapping("/v1/surveys")
    public ResponseEntity<NoDataResponse> createSurvey(@RequestBody SurveyCreateDTO dto) {

        return ResponseEntity.ok().body(surveyService.createSurvey(dto));
    }

    @PutMapping("/v1/surveys/{surveyId}")
    public ResponseEntity<NoDataResponse> updateSurvey(@RequestBody SurveyUpdateDTO dto, @PathVariable Long surveyId) {
        dto.setSurveyId(surveyId);
        return ResponseEntity.ok().body(surveyService.updateSurvey(dto));
    }

}

File: project/jihochoi-onboarding/QueryBox/src/main/kotlin/com/wlghsp/querybox/ui/api/SurveyRestController.kt
package com.wlghsp.querybox.ui.api

import com.wlghsp.querybox.application.SurveyService
import com.wlghsp.querybox.domain.survey.Survey
import com.wlghsp.querybox.ui.dto.ApiResponse
import com.wlghsp.querybox.ui.dto.SurveyCreateRequest
import com.wlghsp.querybox.ui.dto.SurveyUpdateRequest
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/surveys")
class SurveyRestController(
    private val surveyService: SurveyService,
) {

    @GetMapping("/{id}")
    fun getSurvey(@PathVariable id: Long): ResponseEntity<ApiResponse<Survey>> {
        val survey = surveyService.findSurveyWithQuestionsById(id)
        return ResponseEntity.ok(ApiResponse.success(survey))
    }

    @PostMapping
    fun createSurvey(@RequestBody request: SurveyCreateRequest): ResponseEntity<Long> {
        val id = surveyService.create(request)
        return ResponseEntity.ok(id)
    }

    @PatchMapping("{id}")
    fun updateSurvey(@PathVariable id: Long, @RequestBody request: SurveyUpdateRequest): ResponseEntity<Unit> {
        surveyService.update(id, request)
        return ResponseEntity.ok().build()
    }

}
